jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/file/AssetFileGenerator.java

```diff
@@ -219,6 +221,8 @@ public void generateFiles(List<SubscriptionVH> subscriptions, String filePath) {
 		try {
 			FileManager.initialise(filePath);
 		} catch (IOException e1) {
+			log.error("Error occurred in job azure-data-collector-job. Failed to create file in S3 in the given path");
+			AzureDiscoveryJob.jobStatus = "failed";
```

Comment:
again, it's `public static`. I commented about it already - same problem. Are you 100% sure that this `AzureDiscoveryJob` will have only single instance per job execution all the time? What if we will have a multi-tenancy developed soon - will it be the problem? Let's say we run multiply Azure Discovery jobs in the same time.

Reply:
Suggestion: instead you can change signature of this whole method so it will be returning response (status) object.
Status could be just overall status of execution (boolean) or some object with boolean field (or fields - to monitor several states of the execution) and list of what happened during file generation.

Reply:
it looks fine. we are doing the same thing in ErrorManagerUtil.uploadError() method. this method is invoked from various places and state is maintained in errorMap object. i created one more such field 'jobStatus'. 

Will update the code as per the new approaches implemented in future.